modules/migrate.py

```
@@ -0,0 +1,485 @@
+                if os.path.exists(path):
+                    newPath = os.path.relpath(path, modularWidgetAbsolutePath)
+
+                    # Target specifically the parameter with a relative path
+                    # inside the <object> tag we are migrating. If we didn't
+                    # include the <object> tag we could accidentally change
+                    # a relative path for a widget that wasn't migrating that
+                    # referenced the same asset / content.
+                    toReplace = (r'(<object.*?data="' + relativeModuleViewPath +
+                        '".*?<param.*?value=")' + val + '(".*?)(?=</object>)')
+                    lines = re.sub(toReplace,
+                                   r'\g<1>' + newPath + r'\g<2>',
+                                   lines,
+                                   flags=re.IGNORECASE|re.DOTALL)
+
+                    # If the param is for the widget config file, also clean up
+                    # that config file.
+                    if param['name'] == 'configFile':
+                        _updateConfigFile(path, widgetAbsolutePath,
+                                           modularWidgetAbsolutePath)
+
+        # Write out all the changes.
+        log.info('Writing updated HTML file: %s', filePath)
+        with open(filePath, 'wb') as htmlFile:
+           htmlFile.write(lines)
+
+def _updateConfigFile(filePath, widgetPath, modularWidgetPath):
+    """Updates all values nested in the JSON object that are paths relative to
+    the widgetPath to instead be relative to modularWidgetPath.
+
+        Args:
+            filePath - String absolute path to the JSON config file.
+            widgetPath - String absolute path to the non-modular widget.
+            modularWidthPath - String absolute path to the modular widget.
+    """
+    log.debug('Reading JSON config file: %s', filePath)
+    with open(filePath, 'rb') as configFile:
```

Very torn on this. It's kind of nice, but you're still in the exact same state where you have to manually fix the migration somewhere (config file is still wrong). I'd almost just wait and see how often it happens. I expect if they are problems they are far more likely to be that the scripts do the wrong thing, than that they have file i/o errors. And if we don't just let it propagate, how do we nicely report up the errors. Thoughts?